haystack_experimental/components/query/query_expander.py

```diff
@@ -0,0 +1,272 @@
+# SPDX-FileCopyrightText: 2022-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import json
+from typing import Any, Dict, List, Optional
+
+from haystack import default_from_dict, default_to_dict, logging
+from haystack.components.builders.prompt_builder import PromptBuilder
+from haystack.components.generators.chat.openai import OpenAIChatGenerator
+from haystack.components.generators.chat.types import ChatGenerator
+from haystack.core.component import component
+from haystack.core.serialization import component_to_dict
+from haystack.dataclasses.chat_message import ChatMessage
+from haystack.utils.deserialization import deserialize_chatgenerator_inplace
+
+logger = logging.getLogger(__name__)
+
+
+DEFAULT_PROMPT_TEMPLATE = """
+You are part of an information system that processes user queries for retrieval.
+You have to expand a given query into {{ n_expansions }} queries that are
+semantically similar to improve retrieval recall.
+
+Structure:
+Follow the structure shown below in examples to generate expanded queries.
+
+Examples:
+1.  Query: "climate change effects"
+    {"queries": ["impact of climate change", "consequences of global warming", "effects of environmental changes"]}
+
+2.  Query: "machine learning algorithms"
+    {"queries": ["neural networks", "clustering techniques", "supervised learning methods", "deep learning models"]}
+
+3.  Query: "open source NLP frameworks"
+    {"queries": ["natural language processing tools", "free nlp libraries", "open-source NLP platforms"]}
+
+Guidelines:
+- Generate queries that use different words and phrasings
+- Include synonyms and related terms
+- Maintain the same core meaning and intent
+- Make queries that are likely to retrieve relevant information the original might miss
+- Focus on variations that would work well with keyword-based search
+- Respond in the same language as the input query
+
+Your Task:
+Query: "{{ query }}"
+
+You *must* respond with a JSON object containing a "queries" array with the expanded queries.
+Example: {"queries": ["query1", "query2", "query3"]}"""
+
+
+@component
+class QueryExpander:
```

Comment:
Seems like we are missing a `warm_up` method. Could you check out this [example](https://github.com/deepset-ai/haystack/blob/1c7ee927f070d7fda7b25a2283eb0ceb4639fd48/haystack/components/routers/llm_messages_router.py#L93-L100) from @anakin87 on another component that takes in a ChatGenerator as input?

Reply:
Added `warm_up()` support and tests.